hawkular-inventory-api/src/main/java/org/hawkular/inventory/api/model/Change.java

```diff
@@ -0,0 +1,90 @@
+/*
+ * Copyright 2015-2016 Red Hat, Inc. and/or its affiliates
+ * and other contributors as indicated by the @author tags.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package org.hawkular.inventory.api.model;
+
+import java.time.Instant;
+
+import org.hawkular.inventory.api.Action;
+
+/**
+ * Represents a change made to an entity at some point in time.
+ *
+ * @author Lukas Krejci
+ * @since 0.19.0
+ */
+public final class Change<Element extends AbstractElement<?, ?>, Context> implements Comparable<Change<?, ?>> {
+    private final Instant time;
+    private final Action<Context, Element> action;
+    private final Context actionContext;
+    private final Element element;
+
+    public Change(Instant time, Action<Context, Element> action, Context actionContext, Element element) {
+        this.time = time;
+        this.action = action;
+        this.actionContext = actionContext;
+        this.element = element;
+    }
+
+    public Instant getTime() {
+        return time;
+    }
+
+    public Action<Context, Element> getAction() {
+        return action;
+    }
+
+    public Context getActionContext() {
+        return actionContext;
+    }
+
+    public Element getElement() {
+        return element;
+    }
+
+    @Override public int compareTo(Change<?, ?> o) {
+        int diff = time.compareTo(o.time);
```

Comment:
Yes, I put my comment here because that's where I noticed potential NPE but the null-checks, if we decide to do them, should definitely be done in constructors or setters.

It's not a remark for this PR in particular, but more a general thought.